File: Bai_107/main.cpp
#include<bits/stdc++.h>
using namespace std;

using ll = long long;

bool isFibonacci(int n){
    if(n == 0 and n == 1) return true;
    int a = 0, b = 1;
    while(b < n){
        int tmp = b;
        b += a;
        a = tmp;
    }
    return b == n;
}

int main(){
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int n, sum = 0; cin >> n;
    for(int i = 0; i < n; ++i){
        int num; cin >> num;
        if(isFibonacci(num))
            sum += num;
    }
    return 0;
}

File: Bai_107/main.py
def is_fibonacci(n):
    if n == 0 or n == 1:
        return True
    a, b = 0, 1
    while b < n:
        a, b = b, a + b
    return b == n

n = int(input())
arr = list(map(int, input().split()))
s = 0

for num in arr:
    if is_fibonacci(num):
        s += num
        
print(s)
